### PR TITLE
etcdserver: stats/server - refactored

### PR DESCRIPTION
**Pull request by sinsharat**
removed code duplicacy and improved readability

### CODE REVIEW

File: etcdserver/stats/server.go [hunk 1]
```diff
@@ -145,9 +141,14 @@ func (ss *ServerStats) BecomeLeader() {
 	ss.Lock()
 	defer ss.Unlock()
 
+	ss.becomeLeader(time.Now())
+}
+
+func (ss *ServerStats) becomeLeader(now time.Time) {
+
```
Comment (xiang90) [inline etcdserver/stats/server.go]: remove this empty line?

File: etcdserver/stats/server.go [hunk 2]
```diff
@@ -145,9 +141,14 @@ func (ss *ServerStats) BecomeLeader() {
...
 	if ss.State != raft.StateLeader {
 		ss.State = raft.StateLeader
 		ss.LeaderInfo.Name = ss.ID
-		ss.LeaderInfo.StartTime = time.Now()
+		ss.LeaderInfo.StartTime = now
```
Comment (xiang90) [inline etcdserver/stats/server.go]: keep time.Now() here? and remove the arg in the method?

File: etcdserver/stats/server.go (hunk 1 above)
Comment (sinsharat) [inline etcdserver/stats/server.go]: ok will do it.

File: etcdserver/stats/server.go (hunk 2 above)
Comment (sinsharat) [inline etcdserver/stats/server.go]: ok sure will do it.

**Pull request closed without merging**